### PR TITLE
Unify graph partitioning scripts and add predict-then-propagate example

### PR DESCRIPTION
**Pull request by bgawrych**
- Replaced partition_mag and partition_arxiv_products scripts with single partition_graph script
- Add support for cora, citeseer and pubmed graphs in partition_graph script
- Add example of using predict-then-propagate model

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
+The ``partition_graph.py`` script can be used to partition both homogeneous and heterogeneous graphs. It utilizes DGL's metis-based partitioning algorithm to divide the graphs into smaller partitions. Note that all node-related information must be included in the graph's ``ndata`` dictionary so that they are correctly partitioned with the graph.
+Similarly, edge-related information must be included in the graph's ``edata`` dictionary
 
-``partition_mag.py`` partitions the [ogbn-mag](https://ogb.stanford.edu/docs/nodeprop/#ogbn-mag) heterogeneous graph. Again, all node-related information are included in the graph's ``ndata`` for the relevant node types
+### Supported datasets:
+- ogbn-products, ogbn-arxiv, ogb-mag from [Open Graph Benchmarks](https://ogb.stanford.edu/)
```
Comment (Kacper-Pietkun) [inline examples/README.md:7]: ogbn-mag

Comment (Kacper-Pietkun) [inline examples/README.md:7]: ogbn-mag

File: examples/partition_graph.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+parser = ArgumentParser(description="Graph partitioning for ogbn-arxiv and ogbn-products")
+
+parser.add_argument("--dataset-root", type=str, default="./datasets/",
+                    help="The OGB datasets folder")
+
+parser.add_argument("--dataset-name", type=str, default="ogbn-arxiv",
+                    choices=["ogbn-arxiv", "ogbn-products", "ogbn-mag",
+                             "cora", "citeseer", "pubmed"],
+                    help="Dataset name. ogbn-arxiv or ogbn-products")
```
Comment (Kacper-Pietkun) [inline examples/partition_graph.py:48]: Description is outdated

File: examples/README.md (hunk 1 above)
Comment (Kacper-Pietkun) [inline examples/README.md:7]:
```suggestion
- ogbn-products, ogbn-arxiv, ogbn-mag from [Open Graph Benchmarks](https://ogb.stanford.edu/)
```

File: examples/partition_graph.py (hunk 2 above)
Comment (bgawrych) [reply examples/partition_graph.py:48]:
```suggestion
                    help="Name of the dataset to partition.")
```

**Review by Kacper-Pietkun — APPROVED** [review]

**Pull request merged**